react/features/local-recording/controller/RecordingController.js

```diff
@@ -0,0 +1,493 @@
+     */
+    getParticipantsStats() {
+        const members
+            = this._conference.getParticipants()
+            .map(member => {
+                return {
+                    id: member.getId(),
+                    displayName: member.getDisplayName(),
+                    recordingStats:
+                        JSON.parse(member.getProperty(PROPERTY_STATS) || '{}'),
+                    isSelf: false
+                };
+            });
+
+        // transform into a dictionary,
+        // for consistent ordering
+        const result = {};
+
+        for (let i = 0; i < members.length; ++i) {
+            result[members[i].id] = members[i];
+        }
+        const localId = this._conference.myUserId();
+
+        result[localId] = {
+            id: localId,
+            displayName: i18next.t('localRecording.localUser'),
+            recordingStats: this.getLocalStats(),
+            isSelf: true
+        };
+
+        return result;
+    }
+
+    _updateStats: () => void;
+
+    /**
+     * Sends out updates about the local recording stats via XMPP.
+     *
+     * @private
+     * @returns {void}
+     */
+    _updateStats() {
+        if (this._conference) {
+            this._conference.setLocalParticipantProperty(PROPERTY_STATS,
+                JSON.stringify(this.getLocalStats()));
+        }
+    }
+
+    _onStartCommand: (*) => void;
+
+    /**
+     * Callback function for XMPP event.
+     *
+     * @private
+     * @param {*} value - The event args.
+     * @returns {void}
+     */
+    _onStartCommand(value) {
+        const { sessionToken, format } = value.attributes;
+
+        if (this._state === ControllerState.IDLE) {
+            this._format = format;
+            this._currentSessionToken = sessionToken;
+            this._adapters[sessionToken]
+                 = this._createRecordingAdapter();
+            this._doStartRecording();
+        } else if (this._currentSessionToken !== sessionToken) {
+            // we need to restart the recording
+            this._doStopRecording().then(() => {
+                this._format = format;
+                this._currentSessionToken = sessionToken;
+                this._adapters[sessionToken]
+                    = this._createRecordingAdapter();
+                this._doStartRecording();
+            });
+        }
+    }
+
+    _onStopCommand: (*) => void;
+
+    /**
+     * Callback function for XMPP event.
+     *
+     * @private
+     * @param {*} value - The event args.
+     * @returns {void}
+     */
+    _onStopCommand(value) {
+        if (this._state === ControllerState.RECORDING
+            && this._currentSessionToken === value.attributes.sessionToken) {
+            this._doStopRecording();
+        }
+    }
+
+    /**
+     * Generates a token that can be used to distinguish each
+     * recording session.
+     *
+     * @returns {number}
+     */
+    _getRandomToken() {
+        return Math.floor(Math.random() * 10000) + 1;
```

Comment:
I suppose if you want to do a random token and not key off of some existing value, I wonder if Math.random is random enough. Maybe checking for uniqueness is also warranted or changing this bit to be less likely to conflict with an existing value.

Reply:
Like GUIDs?

Reply:
Well. Just make it larger.